src/Sql/Predicate/Predicate.php

```diff
@@ -250,7 +250,7 @@ public function notLike($identifier, $notLike)
     public function expression($expression, $parameters = null)
     {
         $this->addPredicate(
-            new Expression($expression, $parameters),
+            new Expression($expression, func_num_args() > 1 ? $parameters : []),
```

Comment:
If I invoke this like: `$predicate->expression('someExpression()', null)` will `func_num_args` be 2 in that case? And will it still break?

If it broke and before and breaks still:
a) do we need to add a test to ensure we get `[null]` in that case?
b) does that also need fixing?

Reply:
> will `func_num_args()` be 2 in that case? 

Yes

 > do we need to add a test to ensure we get `[null]` in that case?

One of the tests added to this patch ensures that `[null]` is preserved

 > does that also need fixing?

No, I'd say that somebody using `->expression($string, $parameter)` is doing so intentionally, and we should leave it alone :)